File: tests/nodejs/cases/case_189_serialized_copy_only_09.js
/**
 * Task case 189: serialized_copy_only_09 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case189SerializedCopyOnly09(input) {
  const taskName = 'serialized_copy_only_09';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const serialized = JSON.stringify(payload);
  // SAFE: only primitive serialized data leaves scope.
  return serialized;
}

module.exports = {
  case189SerializedCopyOnly09,
};

File: tests/go/cases/case_189_serialized_copy_only_09.go
package escape_tests

var retainedCase189 = []map[string]string{}

func Case189SerializedCopyOnly09(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "serialized_copy_only_09",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "serialized_copy_only_09:" + raw,
	}
	serialized := payload["task"] + ":" + payload["input"]
	// SAFE: only primitive string leaves function.
	return serialized
}
